### PR TITLE
Adds 'unused' hints for MDS io_info parameters

### PR DESCRIPTION
**Pull request by derobins**

### CODE REVIEW

**Review by byrnHDF — APPROVED** [review]

**Comment by derobins** [comment]
This is for sure going to fail the formatting check.

For one of the calls, I opted to use (void) instead of an ATTR_UNUSED macro. We don't have one for "used in debug parallel only" and those macros are already ridiculous.

**Comment by derobins** [comment]
Once this is merged, I'll fix the GitHub actions that are incorrectly using the default build mode.

File: src/H5Dchunk.c [hunk 1]
```diff
@@ -2466,7 +2466,12 @@ H5D__chunk_cacheable(const H5D_io_info_t *io_info, H5D_dset_io_info_t *dset_info
     FUNC_ENTER_PACKAGE
 
     /* Sanity check */
+#ifdef H5_HAVE_PARALLEL
     HDassert(io_info);
+#else
+        (void)
+        io_info;
```
Comment (derobins) [inline src/H5Dchunk.c]: lol - yay, formatter

Comment (byrnHDF) [reply src/H5Dchunk.c]: strange -  I wonder if there is a setting for that?

**Review by jhendersonHDF — APPROVED** [review]

File: src/H5Dchunk.c [hunk 2]
```diff
@@ -2466,7 +2466,13 @@ H5D__chunk_cacheable(const H5D_io_info_t *io_info, H5D_dset_io_info_t *dset_info
     FUNC_ENTER_PACKAGE
 
     /* Sanity check */
+#ifdef H5_HAVE_PARALLEL
     HDassert(io_info);
+#else
+        /* clang-format off */
```
Comment (derobins) [inline src/H5Dchunk.c]: wtf? Must be due to the FUNC_ENTER macro.

Comment (derobins) [reply src/H5Dchunk.c]: This is less ugly, and I have zero interest in adding to the already-over-complicated ATTR_UNUSED scheme

**Review by byrnHDF — APPROVED** [review]

**Pull request merged**